test/units/parsing/test_data_loader.py

```diff
@@ -62,6 +63,119 @@ def test_parse_fail_from_file(self, mock_def):
         """, True)
         self.assertRaises(AnsibleParserError, self._loader.load_from_file, 'dummy_yaml_bad.txt')
 
+
+class TestPathDwimDataLoader(unittest.TestCase):
+
+    def setUp(self):
+        self._loader = DataLoader()
+
+    def test_slash(self):
+        ret = self._loader.path_dwim('/')
+        assert ret is not None, "Returned path should not be None"
+        self.assertEqual(ret, '/')
+
+    def test_tilde(self):
+        ret = self._loader.path_dwim('~')
+        assert ret is not None, "Returned path should not be None"
+        self.assertEquals(os.path.expanduser(u'~'), ret)
+
+    def test_tilde_slash(self):
+        ret = self._loader.path_dwim('~/')
+        assert ret is not None, "Returned path should not be None"
+        self.assertEquals(os.path.expanduser(u'~/'), os.path.join(ret, ''))
+
+    def test_dot(self):
+        ret = self._loader.path_dwim('.')
+        assert ret is not None, "Returned path should not be None"
+        self.assertEquals(os.path.abspath(os.path.curdir), ret)
+
+
+class TestPathDwimRelativeDataLoader(unittest.TestCase):
+
+    def setUp(self):
+        self._loader = DataLoader()
+
+    def test_none(self):
+        ret = self._loader.path_dwim_relative(None, None, None)
+        assert ret is not None, "Returned path should not be None"
+        print(ret)
+
+    def test_empty_strings(self):
+        ret = self._loader.path_dwim_relative('', '', '')
+        assert ret is not None, "Returned path should not be None"
+        # FIXME: Not sure what this is expected to do outside of a playbook
+        self.assertEquals(os.path.curdir, ret)
+
+    def test_all_slash(self):
+        ret = self._loader.path_dwim_relative('/', '/', '/')
+        self.assertEquals(u'/', ret)
+
+    def test_path_endswith_role_from_slash(self):
+        ret = self._loader.path_dwim_relative(path='foo/bar/tasks/', dirname='/', source='/')
+        assert ret is not None, "Returned path should not be None"
+        self.assertEquals(u'/', ret)
```

Comment:
this would not be the expected return, it should be a list with one item, `foo/bar/tasks`